numpy/core/code_generators/ufunc_docstrings.py

```diff
@@ -1835,6 +1835,18 @@ def add_newdoc(place, name, doc):
     >>> np.left_shift(5, [1,2,3])
     array([10, 20, 40])
 
+    Note: As explained in :ref:`Casting Rules <ufuncs.casting>`, if a ufunc
+    does not have a core loop implementation for the input types provided,
+    it will cast one or more inputs to a different type.  This may lead to
+    an unexpected result in some cases.
```

Comment:
I think part of what causes the confusion here is also the fact that the second arguments dtype feels like it should not have any effect, but it does.  On the implementation side you are completely right, but I am curious if we can phrase it that way and if that may be easier to understand?

Reply:
I agree.  It should be made easier to understand and the link to the Casting Rules points people in the right direction if they want to go deeper.  I modified the note accordingly.  Let me know how it sounds.